<a href="https://colab.research.google.com/github/jamesharrison2005/CleanedAmazonDataAnalysis/blob/main/Cleaned_Amazon_Sales_Analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Introduction
Brief and basic analysis of kaggle Cleaned amazon sales dataset that can be found here: https://www.kaggle.com/datasets/nadaarfaoui/cleaned-amazon-sales-and-reviews-dataset/data



In [30]:
import pandas as pd
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [31]:
df = pd.read_csv("/content/drive/MyDrive/Colab Notebooks/cleaned_dataset.csv")


## Understanding the Data
Using Pandas to get an idea of how the data is formatted, the amount of rows and columns as well as there names.

In [51]:
df.info()
df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4854 entries, 0 to 4853
Data columns (total 13 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   name                 4854 non-null   object 
 1   image                4854 non-null   object 
 2   link                 4854 non-null   object 
 3   no_of_ratings        4854 non-null   float64
 4   discount_price       4854 non-null   float64
 5   actual_price         4854 non-null   float64
 6   rating               4854 non-null   float64
 7   review_text          4854 non-null   object 
 8   brand                4854 non-null   object 
 9   cleaned_review_text  4826 non-null   object 
 10  sentiment            4854 non-null   object 
 11  is_discounted        4854 non-null   bool   
 12  estimated_revenue    4854 non-null   float64
dtypes: bool(1), float64(5), object(7)
memory usage: 459.9+ KB


name  \
0  OnePlus Nord CE 2 Lite 5G Blue Tide 6GB RAM 12...   
1  OnePlus Bullets Z2 Bluetooth Wireless in Ear E...   
2  Samsung Galaxy M33 5G Mystique Green 6GB 128GB...   
3  OnePlus Nord CE 2 Lite 5G Black Dusk 6GB RAM 1...   
4  Redmi 10 Power Sporty Orange 8GB RAM 128GB Sto...   

                                               image  \
0  https://m.media-amazon.com/images/I/71AvQd3Vzq...   
1  https://m.media-amazon.com/images/I/51UhwaQXCp...   
2  https://m.media-amazon.com/images/I/81I3w4J6yj...   
3  https://m.media-amazon.com/images/I/71V--WZVUI...   
4  https://m.media-amazon.com/images/I/81OZGH4fZi...   

                                                link  no_of_ratings  \
0  https://www.amazon.in/OnePlus-Nord-Lite-128GB-...       113956.0   
1  https://www.amazon.in/Oneplus-Bluetooth-Wirele...        90304.0   
2  https://www.amazon.in/Samsung-Mystique-Storage...        24863.0   
3  https://www.amazon.in/OnePlus-Nord-Black-128GB...       113956.0   
4  https://www.amazon.in/Redmi-Power-Sporty-Orang...          625.0   

   discount_price  actual_price  rating  \
0         18999.0       19999.0     1.0   
1          1999.0        2299.0     5.0   
2         15999.0       24999.0     1.0   
3         18999.0       19999.0     5.0   
4         10999.0       18999.0     1.0   

                                         review_text    brand  \
0  Very bad experience with this device xr phone....  OnePlus   
1  Amazing phone with amazing camera coming from ...  OnePlus   
2  So I got the device XR just today. The product...  Samsung   
3  I've been an android user all my life until I ...  OnePlus   
4  I was delivered a phone that did not work imme...    Redmi   

                                 cleaned_review_text sentiment  is_discounted  \
0  bad experience device xr phone back camera fou...  Negative           True   
1  amazing phone amazing camera coming device plu...  Positive           True   
2  got device xr today product look amazing unfor...  Negative           True   
3  ive android user life decided try device xr io...  Positive           True   
4  delivered phone work immediately removing box ...  Negative           True   

   estimated_revenue  
0       2.165050e+09  
1       1.805177e+08  
2       3.977831e+08  
3       2.165050e+09  
4       6.874375e+06

In [33]:
df.describe()

no_of_ratings  discount_price   actual_price       rating
count    4854.000000     4854.000000    4854.000000  4854.000000
mean    13212.038662     3071.930556    5197.628795     4.535435
std     34629.373129     8369.731769   11818.224992     1.070116
min         1.000000       36.000000      35.000000     1.000000
25%       519.000000      299.000000     837.250000     5.000000
50%      2360.000000      713.000000    1499.000000     5.000000
75%     10322.000000     1999.000000    3999.000000     5.000000
max    437651.000000   127999.000000  149999.000000     5.000000

# Data Exploration
4854 data entries with 11 columns all of which are not nullable. Datasets consist of objects and floats. Features of the dataset revolve around product information, as well as discounted and actual prices, and reviews of the products along with the sentiment of the review.

##Sales Performance Analysis
Grouping features together can help provide insight on the performance of certain products which can allow us to find answers to important questions such as:
- Which products have the highest amount of positive ratings?
- which products have the highest estimated revenue?
- Which Brands bring have the highest ratings?

### Products with the Most Positive Ratings
To determine which products have the highest amount of positive ratings, we filter the DataFrame for 'Positive' sentiment and then sum the 'no_of_ratings' for each product.

In [48]:
positive_ratings_by_product = df[df['sentiment'] == 'Positive'].groupby('name')['no_of_ratings'].sum().sort_values(ascending=False)
print("Products with the most positive ratings:\n", positive_ratings_by_product.head(10))

Products with the most positive ratings:
 name
pTron Bassbuds Duo in Ear Earbuds with 32Hrs Total Playtime Bluetooth 51 Wireless Stereo Audio Touch Control TWS with       499185.0
Amazon Basics Flexible Premium HDMI Cable Black 4K60Hz 18Gbps 3Foot                                                         437651.0
Amazon Basics HighSpeed HDMI Cable  10 Feet Latest Standard  Supports Ethernet 3D 4K videoBlack                             437651.0
Amazon Basics HighSpeed HDMI Cable 6 Feet  Supports Ethernet 3D 4K videoBlack                                               437651.0
Amazon Basics HighSpeed HDMI Cable 6 Feet 2PackBlack                                                                        437651.0
boAt Airdopes 121v2 inEar True Wireless Earbuds with Upto 14 Hours Playback 8MM Drivers Battery Indicators Lightweight      376062.0
boAt Airdopes 121 PRO True Wireless in Ear Earbuds Signature Sound Quad Mic ENx Beast Mode for Gaming 40H Playtime IW       376062.0
boAt Bassheads 100 Wir

###Products with the highest estimated revenue


In [55]:
highest_revenue_products = df.groupby('name')['estimated_revenue'].sum().sort_values(ascending=False)
display(highest_revenue_products)

name
Redmi Note 10S Deep Sea Blue 6GB RAM 64GB Storage  Super Amoled Display  64 MP Quad Camera  6 Month Free Screen Rep        3.318814e+09
Redmi 9 Sporty Orange 4GB RAM 64GB Storage  23GHz Mediatek Helio G35 Octa core Processor                                   3.190766e+09
Redmi 9A Sport Carbon Black 3GB RAM 32GB Storage  2GHz Octacore Helio G25 Processor  5000 mAh Battery                      2.395151e+09
Redmi 9A Sport Coral Green 3GB RAM 32GB Storage  2GHz Octacore Helio G25 Processor  5000 mAh Battery                       2.395151e+09
OnePlus Nord CE 2 Lite 5G Blue Tide 8GB RAM 128GB Storage                                                                  2.392962e+09
                                                                                                                               ...     
TheGiftKart Shockproof Crystal Clear Back Cover Case for Vivo V27  V27 PRO 5G  360 Degree Protection  Protective Design    1.990000e+02
OUXUS USB all types smartwatch charger cable for boat watch charger  boat xtend watch charger  boat storm watch charger    1.980000e+02
Fashionury Back Case Cover for Samsung Galaxy S21 FE 5G  Crystal Clear Hard Back  Shockproof Design  Camera Protection     1.490000e+02
Onsitego 2 Year Extended Warranty for Small Household Appliances from Rs 10011500 Email Delivery  No Physical Kit          1.250000e+02
Onsitego 1 Year Extended Warranty for Small Household Appliances from Rs 10011500 Email Delivery  No Physical Kit          9.500000e+01
Name: estimated_revenue, Length: 4467, dtype: float64

###Highest estimated revenue with formatting

In [58]:
formatted_highest_er = df.groupby('name')['estimated_revenue'].sum().apply(lambda x: f"{x / 1_000_000_000:.2f} billion").sort_values(ascending=False)
display(formatted_highest_er.head(10))

name
Redmi Note 10S Deep Sea Blue 6GB RAM 64GB Storage  Super Amoled Display  64 MP Quad Camera  6 Month Free Screen Rep    3.32 billion
Redmi 9 Sporty Orange 4GB RAM 64GB Storage  23GHz Mediatek Helio G35 Octa core Processor                               3.19 billion
Redmi 9A Sport Carbon Black 3GB RAM 32GB Storage  2GHz Octacore Helio G25 Processor  5000 mAh Battery                  2.40 billion
Redmi 9A Sport Coral Green 3GB RAM 32GB Storage  2GHz Octacore Helio G25 Processor  5000 mAh Battery                   2.40 billion
OnePlus Nord CE 2 Lite 5G Blue Tide 8GB RAM 128GB Storage                                                              2.39 billion
Samsung Galaxy M12 Black6GB RAM 128GB Storage 6 Months Free Screen Replacement for Prime                               2.39 billion
OnePlus Nord CE 2 Lite 5G Black Dusk 6GB RAM 128GB Storage                                                             2.17 billion
OnePlus Nord CE 2 Lite 5G Blue Tide 6GB RAM 128GB Storage                                                              2.17 billion
Redmi 9A Sport Carbon Black 2GB RAM 32GB Storage  2GHz Octacore Helio G25 Processor  5000 mAh Battery                  2.08 billion
Redmi 9A Sport Coral Green 2GB RAM 32GB Storage  2GHz Octacore Helio G25 Processor  5000 mAh Battery                   2.08 billion
Name: estimated_revenue, dtype: object